examples/C++/DDS/BasicConfigurationExample/BasicConfigurationPublisher.cpp

```diff
@@ -0,0 +1,284 @@
+#include <fastdds/rtps/transport/UDPv4TransportDescriptor.h>
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/PublisherAttributes.h>
+
+#include "BasicConfigurationPublisher.h"
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastdds::rtps;
+
+std::atomic<bool> HelloWorldPublisher::stop_(false);
+std::mutex HelloWorldPublisher::PubListener::wait_matched_cv_mtx_;
+std::condition_variable HelloWorldPublisher::PubListener::wait_matched_cv_;
+
+HelloWorldPublisher::HelloWorldPublisher()
+    : participant_(nullptr)
+    , publisher_(nullptr)
+    , topic_(nullptr)
+    , writer_(nullptr)
+    , type_(new HelloWorldPubSubType())
+{
+}
+
+bool HelloWorldPublisher::is_stopped()
+{
+    return stop_;
+}
+
+void HelloWorldPublisher::stop()
+{
+    stop_ = true;
+    PubListener::awake();
+}
+
+bool HelloWorldPublisher::init(
+        const std::string& topic_name,
+        uint32_t domain,
+        uint32_t num_wait_matched,
+        bool async,
+        const std::string& transport,
+        bool reliable,
+        bool transient)
+{
+    hello_.index(0);
+    hello_.message("HelloWorld");
+    DomainParticipantQos pqos;
+    pqos.name("Participant_pub");
+    listener_.set_num_wait_matched(num_wait_matched);
+
+    // TRANSPORT CONFIG
+    if (!transport.empty())
+    {
+        pqos.transport().use_builtin_transports = false;
+        if (transport == "shm")
+        {
+            auto shm_transport = std::make_shared<SharedMemTransportDescriptor>();
+            pqos.transport().user_transports.push_back(shm_transport);
+        }
+        else
+        {
+            auto udp_transport = std::make_shared<UDPv4TransportDescriptor>();
+            pqos.transport().user_transports.push_back(udp_transport);
+        }
+    }
+
+    // CREATE THE PARTICIPANT
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(domain, pqos);
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    // REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    // CREATE THE PUBLISHER
+    publisher_ = participant_->create_publisher(PUBLISHER_QOS_DEFAULT, nullptr);
+
+    if (publisher_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE TOPIC
+    topic_ = participant_->create_topic(topic_name, "HelloWorld", TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE WRITER
+    DataWriterQos wqos = DATAWRITER_QOS_DEFAULT;
+    if (!transport.empty())
+    {
+        wqos.data_sharing().off();
+    }
+    else
+    {
+        wqos.data_sharing().automatic();  // default
+    }
+
+    if (async)
+    {
+        wqos.publish_mode().kind = ASYNCHRONOUS_PUBLISH_MODE;
+    }
+    else
+    {
+        wqos.publish_mode().kind = SYNCHRONOUS_PUBLISH_MODE;    // default
+    }
+
+    if (reliable)
+    {
+        wqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
```

Comment:
Add a comment
```suggestion
        wqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
        wqos.reliability().kind = KEEP_ALL_HISTORY_QOS;
```

Reply:
:+1:
Added this line also to durability part.